plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java

```diff
@@ -43,23 +44,23 @@ protected AbstractControl(T swtWidget, Runnable diagnostics) {
 	 * @return true if widget is enabled, false otherwise
 	 */
 	public boolean isEnabled(){
-		return controlHandler.isEnabled(swtWidget);
+		return ControlHandler.getInstance().isEnabled(swtWidget);
 	}
 	
 	public boolean isVisible(){
-		return controlHandler.isVisible(swtWidget);
+		return ControlHandler.getInstance().isVisible(swtWidget);
 	}
 	
 	public void setFocus(){
-		controlHandler.setFocus(swtWidget);
+		ControlHandler.getInstance().setFocus(swtWidget);
 	}
 	
 	public boolean isFocusControl(){
```

Comment:
this comes from eclipse swt Control API. There is not hasFocus but isFocusControl. We can discuss changing it to hasFocus in some future improvements